src/OperationStore.ts

```diff
@@ -1,89 +1,164 @@
-import BatchFile from './BatchFile';
-import Logger from './lib/Logger';
-import { Cache, getCache } from './Cache';
-import { Cas } from './Cas';
-import { Operation } from './Operation';
-import * as startTimer from 'time-span';
+import { getOperationHash, OperationType, Operation } from './Operation';
 
 /**
- * Information of an operation that is required to reconstruct it from
- * the CAS.
+ * An abstraction of a complete store for operations exposing methods to
+ * put and get operations.
  */
-interface OperationAccessInfo {
-  readonly batchFileHash: string;
-  readonly operationIndex: number;
-  readonly transactionTime: number;
-  readonly transactionNumber: number;
+export interface OperationStore {
+
+  /**
+   * Store an operation.
+   */
+  put (operation: Operation): Promise<void>;
+
+  /**
+   * Get an iterator that returns all operations with a given
+   * did ordered by (transactionNumber, operationIndex)
+   * ascending.
+   */
+  get (did: string): Promise<Iterable<Operation>>;
```

Comment:
Rename `did` param and in the comment to `didUniqueSuffix` to disambiguate.

Reply:
Done.